packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/CameraCaptureCallback.java

```diff
@@ -67,6 +72,20 @@ private void process(CaptureResult result) {
     Integer aeState = result.get(CaptureResult.CONTROL_AE_STATE);
     Integer afState = result.get(CaptureResult.CONTROL_AF_STATE);
 
+    // Update capture properties
+    Float lensAperture = result.get(CaptureResult.LENS_APERTURE);
+    Long sensorExposureTime = result.get(CaptureResult.SENSOR_EXPOSURE_TIME);
+    Integer sensorSensitivity = result.get(CaptureResult.SENSOR_SENSITIVITY);
+    if (lensAperture != null) {
+      this.captureProps.setLastLensAperture(lensAperture);
+    }
```

Comment:
Quick check, is the `null` check on purpose? I can imagine that `null` indicates the value was not available for this particular capture result and you want to communicate this back to the Dart side. 

Don't you now have the risk that you communicate a (invalid) previous value with the current image?

(this question also goes for the `sensorExposureTime` and `sensorSensitivity` values)

Reply:
The way I had it in my head, is that in case you have some frames which do not supply these values, there's a very good chance the values will have been the same as the one from the frame(s) before it. Not sure if this is actually the case, but if you have a scenario where the camera implementation only reports back these values every X frames, there wouldn't be any issues.

I do agree that this might be confusing and potentially unwanted behavior though. I'm not sure either whether my concerns actually have any ground.

Reply:
That would certainly be a simpler solution. I guess I'll just do that, if it turns out to give some trouble we can always add something like it back in later.

Reply:
Looking at it now I just remembered why I implemented it this way in the first place:

These properties all have to be sent back in the handler given to [`imageStreamReader.setOnImageAvailableListener`](https://github.com/flutter/plugins/pull/4256/files#diff-97f0364e631a66ce678d82ff9906a3fa2f6fc75cfa6c59511d7d848f8c33544eR1049). 
However, I ran into the problem here that I cannot actually access the CaptureResult which contains these values here. 
Instead, the capture result is only available in `CameraCaptureCallback`, which is where I save these values. 

This is why I opted to cache these, as I don't really see any way of associating a specific image with a specific capture result.